Fixed login route redirect for already logged users

lukasaric opened the pull request:
- This PR addresses #804
- It properly redirects already logged user to root route (catalog) who's trying to access login route

lukasaric committed:
Fixed public route access redirect for logged user

lukasaric committed:
Improved auth check utils naming

lukasaric committed:
Removed leftover negation :see_no_evil:

lukasaric committed:
Auth check naming improvement

underscope commented:
@malizeus This assumes that we can't have a public page down the road that logged-in users can access. This also introduces a bug upon clicking on forgot password btn on the profile page.

lukasaric committed:
Redirect logged users on accessing login route

lukasaric committed:
Merge branch 'develop' into feature/logged-user-redirect-fix

lukasaric committed:
Aligned conditions & extracted login redirect

underscope commented on client/router.js:
```diff
@@ -120,20 +120,24 @@ const options = {
-    }
-    return isAllowed(to, auth) ? next() : next({ path: from.fullPath });
+    const { user } = router.app.$store.state.auth;
+    if (isAuthRequired(to, user)) return redirectToLogin(to, next);
```
This might be a bit clearer if isAuthRequired would just check is auth required and not check if a user is logged in 😉  💅

underscope commented on client/router.js:
```diff
@@ -120,20 +120,24 @@ const options = {
+    const { user } = router.app.$store.state.auth;
+    if (isAuthRequired(to, user)) return redirectToLogin(to, next);
+    if (to.name === 'login' && user) return next({ path: '/' });
+    return !isAllowed(to, user) ? next({ path: from.fullPath }) : next();
```
Why adding the `!` and moving the other condition first 🤔 ?

lukasaric committed:
Conditions polish :lipstick:

- Decoupled logged user check & is auth required
- Reverted is allowed condition
